irc_send.c

```diff
@@ -213,13 +213,7 @@ void irc_send_names(irc_channel_t *ic)
 			*namelist = 0;
 		}
 
-		if (icu->flags & IRC_CHANNEL_USER_OP) {
-			strcat(namelist, "@");
-		} else if (icu->flags & IRC_CHANNEL_USER_HALFOP) {
-			strcat(namelist, "%");
-		} else if (icu->flags & IRC_CHANNEL_USER_VOICE) {
-			strcat(namelist, "+");
-		}
+		namelist[strlen(namelist)] = irc_channel_user_get_prefix(icu);
```

Comment:
OOOHHHhhhh i get your point now. Yeah, my code is certainly doing it wrong here. If the names list fills the 385 byte buffer, it's sent through and only the first byte is set to 0. Starting from the second pass, the buffer definitely holds garbage.

How about

```c
namelist[strlen(namelist) + 1] = '\0'
namelist[strlen(namelist)] = irc_channel_user_get_prefix(icu);
```